yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -290,8 +290,18 @@ private[yarn] class YarnAllocator(
       location: String,
       containersToUse: ArrayBuffer[Container],
       remaining: ArrayBuffer[Container]): Unit = {
+    // SPARK-6050: certain Yarn configurations return a resource configuration that doesn't match
+    // the request; for example, capacity scheduler + DefaultResourceCalculator. Allow users in
+    // those situations to disable resource matching.
+    val matchingResource =
+        if (sparkConf.getBoolean("spark.yarn.container.matchAnyResource", false)) {
```

Comment:
seems like the config is named backwards?  If I want to match any then I want to use allocatedContainer.getResource.

Perhaps matchExactResource or keep the name and switch what you get.  I would expect to match any by default so its backwards compatible for now.